application/classes/Ushahidi/Repository/Post.php

```diff
@@ -106,6 +114,15 @@ protected function getTable()
 	// Ushahidi_Repository
 	public function getEntity(Array $data = null)
 	{
+		// Ensure we are dealing with a structured Post
+		if ($data['form_id'])
+		{
+			$user = $this->getUser();
+			$this->restricted = $this->restrictPostValues(new Post($data), $user, $this->form_repo);
+			// Get Hidden Stage Ids to be excluded from results
+			$this->exclude_stages = $this->form_stage_repo->getHidenStageIds($data['form_id']);
```

Comment:
You could probably actually return the query here, not just the IDs and run the entire thing as a single subquery.